konrad/core.py

```diff
@@ -228,8 +238,28 @@ def is_converged(self):
         Returns:
             bool: ``True`` if converged, else ``False``.
         """
-        # TODO: Implement proper convergence criterion (e.g. include TOA).
-        return np.all(np.abs(self.deltaT) < self.delta)
+        
+        self.newDN = self.newN - self.oldN
+        self.newDDN = np.abs(self.newDN) - np.abs(self.oldDN)
+        
+        test1 = np.abs(self.newDN) <= self.delta
+        test2 = np.abs(self.newDDN) <= (self.delta / 100)
```

Comment:
What would be the difference, if you only checked the latter (`test2`)?

Reply:
This pseudo-second-derivative criterion reduced false iteration stops, which happened near the adjustment peak that appear in the `abrupt nxCO2` experiments. I introduced it before implementing the limit of iterations after reaching convergence.

However, a near-zero second derivative provides an equilibrium condition only if the first derivative is also small. Therefore, only the pseudo-second-derivative criterion is not enough as a stopping condition.